src/c2pa/c2pa.py

```
@@ -257,62 +334,68 @@ class Verify(Exception):
-                raise C2paError.Io(message)
-            elif error_type == "Json":
-                raise C2paError.Json(message)
-            elif error_type == "Manifest":
-                raise C2paError.Manifest(message)
-            elif error_type == "ManifestNotFound":
-                raise C2paError.ManifestNotFound(message)
-            elif error_type == "NotSupported":
-                raise C2paError.NotSupported(message)
-            elif error_type == "Other":
-                raise C2paError.Other(message)
-            elif error_type == "RemoteManifest":
-                raise C2paError.RemoteManifest(message)
-            elif error_type == "ResourceNotFound":
-                raise C2paError.ResourceNotFound(message)
-            elif error_type == "Signature":
-                raise C2paError.Signature(message)
-            elif error_type == "Verify":
-                raise C2paError.Verify(message)
-        if len(error_str) > 0:
-            raise C2paError.Other(error_str)
-        else:
-            raise C2paError("c2pa_error returned empty string")
-    raise C2paError("c2pa_error returned null")
+class _StringContainer:
+    """Container class to hold encoded strings and prevent them from being garbage collected.
+
+    This class is used to store encoded strings that need to remain in memory
+    while being used by C functions. The strings are stored as instance attributes
+    to prevent them from being garbage collected.
+
+    This is an internal implementation detail and should not be used outside this module.
+    """
+    def __init__(self):
+        """Initialize an empty string container."""
+        pass
 
 def _handle_string_result(result: ctypes.c_void_p, check_error: bool = True) -> Optional[str]:
     """Helper function to handle string results from C2PA functions."""
     if not result:  # NULL pointer
         if check_error:
-            _handle_c2pa_error()
+            error = _lib.c2pa_error()
```

This feels odd to me. handle_string_result is designed specifically to return string results on methods that return strings. Using it everywhere when you already know you want to return an error is confusing. why not just call handle_c2pa_error if that is what you know you are doing?

---

There is no `handle_c2pa_error` in that branch... ? Maybe in your branch... (I'll check and update if I see it there)? And I keep having to check `_lib.c2pa_error()` to not miss any error (remember we had the case where an error string would be reported there, but raise no error, or the called function's result would report being ok...). This is the additional layer to not miss any error with the current setup.

---

I definitely want to throw errors when they happen! I created a bunch of odd little macros to handle things reliably in  the Rust C binding logic. They are oddly named and I need to document them and explain how they work. I've tried to follow old school C error handling in the C APIs, but it needs some work. Ideally every function would return an indication if there is an error or not, either by returning a negative number or returning null. If there's an error, we would always read the last error message and then throw a constructed error in the higher level language. The error string returned by c2pa_error is formatted to included an error name followed by a colon. The rest of the string contains the error message. The error name can be converted into a specific kind of C2paError in languages like Python. 
I know there some inconsistencies and I've been trying to eliminate them. I want all possible null parameters checked and a consistent policy on setting the last error so that we never end up with an empty error state.

---

Ok. In that case, I would suggest to merge that PR into your branch, and rework from there (I really don't want the 2 branches to diverge too much).